src/components/duration/normalized.rs

```diff
@@ -2,34 +2,46 @@
 
 use std::ops::Add;
 
-use crate::{options::TemporalRoundingMode, utils, TemporalError, TemporalResult, NS_PER_DAY};
+use crate::{
+    options::TemporalRoundingMode,
+    rounding::{IncrementRounder, Round},
+    TemporalError, TemporalResult, NS_PER_DAY,
+};
 
 use super::{DateDuration, TimeDuration};
 
-const MAX_TIME_DURATION: f64 = 2e53 * 10e9 - 1.0;
+const MAX_TIME_DURATION: i128 = 9_007_199_254_740_991_999_999_999;
+
+// Nanoseconds constants
+
+const NANOSECONDS_PER_MINUTE: f64 = 60.0 * 1e9;
+const NANOSECONDS_PER_HOUR: f64 = 60.0 * 60.0 * 1e9;
 
 // TODO: This should be moved to i128
 /// A Normalized `TimeDuration` that represents the current `TimeDuration` in nanoseconds.
 #[derive(Debug, Clone, Copy, Default, PartialEq, PartialOrd)]
-pub struct NormalizedTimeDuration(pub(crate) f64);
+pub struct NormalizedTimeDuration(pub(crate) i128);
 
 impl NormalizedTimeDuration {
     /// Equivalent: 7.5.20 NormalizeTimeDuration ( hours, minutes, seconds, milliseconds, microseconds, nanoseconds )
     pub(crate) fn from_time_duration(time: &TimeDuration) -> Self {
-        let minutes = time.minutes + time.hours * 60.0;
-        let seconds = time.seconds + minutes * 60.0;
-        let milliseconds = time.milliseconds + seconds * 1000.0;
-        let microseconds = time.microseconds + milliseconds * 1000.0;
-        let nanoseconds = time.nanoseconds + microseconds * 1000.0;
+        // TODO: Determine if there is a loss in precision from casting. If so, times by 1,000 (calculate in picoseconds) than truncate?
+        let mut nanoseconds: i128 = (time.hours * NANOSECONDS_PER_HOUR) as i128;
+        nanoseconds += (time.minutes * NANOSECONDS_PER_MINUTE) as i128;
+        nanoseconds += (time.seconds * 1e9) as i128;
+        nanoseconds += (time.milliseconds * 1e6) as i128;
+        nanoseconds += (time.microseconds * 1_000.0) as i128;
+        nanoseconds += time.nanoseconds as i128;
         // NOTE(nekevss): Is it worth returning a `RangeError` below.
         debug_assert!(nanoseconds.abs() <= MAX_TIME_DURATION);
         Self(nanoseconds)
     }
 
+    // NOTE: `days: f64` should be an integer.
```

Comment:
Is this going to be changed in a follow up? I guess there's too many call sites right now to change in this PR

Reply:
More than likely, yes. I'm pretty sure I wanted to be able to verify in all cases that it would be `days: u64` or some equivalent before making the change.